File: Backend/src/main/java/com/example/demo/Models/CommentModel.java
package com.example.demo.Models;

import java.util.Date;

import com.example.demo.DemoApplication;
import com.fasterxml.jackson.core.JsonProcessingException;

public class CommentModel {

    private Long id ;
    private String content ;
    private String imagePath ;
    private Date createdAt;
    private PostModel postModel;
    private UserModel userModel;

    public CommentModel() {
    }

    public CommentModel(Long id, String content, Date createdAt, PostModel postModel, UserModel userModel) {
        this.id = id;
        this.content = content;
        this.createdAt = createdAt;
        this.postModel = postModel;
        this.userModel = userModel;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }


    public String getImagePath() {
        return this.imagePath;
    }

    public void setImagePath(String imagePath) {
        this.imagePath = imagePath;
    }


    

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public PostModel getPostModel() {
        return postModel;
    }

    public void setPostModel(PostModel postModel) {
        this.postModel = postModel;
    }

    public UserModel getUserModel() {
        return userModel;
    }

    public void setUserModel(UserModel userModel) {
        this.userModel = userModel;
    }

    @Override
    public String toString() {
        try {
            return DemoApplication.objectMapper.writeValueAsString(this);
        } catch (JsonProcessingException e) {   
            e.printStackTrace();
        }
        return null ;
    }
    
    
}

File: Frontend/lib/Model/CommentModel.dart
import 'package:frontend/Model/UserModel.dart';
import 'package:frontend/main.dart';

import 'PostModel.dart';

class CommentModel {
  String? id;
  String? content;
  String? imagePath;
  String? createdAt;
  PostModel? postModel;
  UserModel? userModel;

  CommentModel({
    this.id,
    this.userModel,
    this.postModel,
    this.imagePath,
    this.content,
    this.createdAt,
  });

  CommentModel.fromJson(Map<String, dynamic> json) {
    id = json["id"].toString();
    content = json["content"];
    createdAt = json["createdAt"];
    if (json["imagePath"] != null) imagePath = json["imagePath"];
    if (json["postModel"] != null) {
      postModel = PostModel.fromJson(json["postModel"]);
    }
    if (json["userModel"] != null) {
      userModel = UserModel.fromJson(json["userModel"]);
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['content'] = content;
    data['createdAt'] = createdAt;
    if (imagePath != null) data['imagePath'] = imagePath;
    if (postModel != null) data['postModel'] = postModel!.toJson();
    if (userModel != null) data['userModel'] = userModel!.toJson();
    return data;
  }

  get getId => id;

  set setId(id) => this.id = id;

  get getUserModel => userModel;

  set setUserModel(userModel) => this.userModel = userModel;

  get getPostModel => postModel;

  set setPostModel(postModel) => this.postModel = postModel;

  get getContent => content;

  set setContent(content) => this.content = content;

  get getImage => imagePath;

  set setImage(imagePath) => this.imagePath = imagePath;

  get getCreatedAt => createdAt;

  set setCreatedAt(createdAt) => this.createdAt = createdAt;

  static List<CommentModel> comments = [
    CommentModel(
      id: "1",
      userModel: MyApp.currentUser,
      imagePath: 'images/photo1.jpg',
      content:
          " My name is ghalia sabbagh , I have one sisterhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh and three brothers , I live in Damascus , I am studying IT",
      createdAt: "1/1/2001",
    ),
    CommentModel(
      id: "2",
      userModel: UserModel.users[4],
      imagePath: null,
      content:
          "Sunday\nMonday\nTuesday\nWednesday\nThursday\nFridady\nSaturday",
      createdAt: "1/1/2001",
    ),
    CommentModel(
        id: "3",
        userModel: MyApp.currentUser,
        imagePath: null,
        content: "can i help you \n can i help you \n can i help you \n ",
        createdAt: "1/1/2001"),
    CommentModel(
      id: "4",
      userModel: MyApp.currentUser,
      imagePath: 'images/photo1.jpg',
      content: null,
      createdAt: "1/1/2001",
    ),
    CommentModel(
      id: "5",
      userModel: UserModel.users[2],
      imagePath: 'images/photo1.jpg',
      content: "lloly",
      createdAt: "1/1/2001",
    ),
    CommentModel(
      id: "6",
      userModel: UserModel.users[3],
      imagePath: null,
      content:
          "Listen to the audio without reading the text. Just listen and try to hear how much you can understand.\n\n Do you recognise any words or phrases? Can you understand what they are talking about?",
      createdAt: "1/1/2001",
    ),
  ];
}
